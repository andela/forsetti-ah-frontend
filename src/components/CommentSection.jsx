import React from 'react';
import SingleComment from './SingleComment';

const CommentSection = (props) => {
  const {
    commentsList,
    commentCount,
    slug,
    postNewThreadComment,
    token,
    history
  } = props;
  return (
    <div className='container mb-5'>
      {commentCount > 0 && commentsList.map(comment => (
        <SingleComment
          key={`${comment.id}${comment.createdAt}`}
          slug={slug}
          postNewThreadComment={postNewThreadComment}
          history={history}
          token={token}
          {...comment}
        />
      ))}
    </div>
  );
};

export default CommentSection;
