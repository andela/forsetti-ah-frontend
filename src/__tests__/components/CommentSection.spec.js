import React from 'react';
import { shallow } from 'enzyme';
import CommentSection from '../../components/CommentSection';

const props = {
  commentsList: [{ id: 1 }],
  commentCount: 1
};

let wrapper;

describe('<CommentSection />', () => {
  test('should render comment component correctly', () => {
    const wrapper = shallow(<CommentSection />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should contain comments if any', () => {
    wrapper = shallow(<CommentSection {...props} />);
    expect(wrapper.exists('SingleComment')).toEqual(true);
  });
});
