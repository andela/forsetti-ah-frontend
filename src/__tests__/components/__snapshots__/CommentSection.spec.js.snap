// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentSection /> should render comment component correctly 1`] = `ShallowWrapper {}`;
